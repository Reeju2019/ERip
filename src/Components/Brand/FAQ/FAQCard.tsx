import React, { useState } from 'react'
import { Collapse } from 'react-bootstrap'

interface IFAQCardProps {
  faq: {
    question: string
    answer: string
    line: boolean
  }
}

const FAQCard: React.FunctionComponent<IFAQCardProps> = (faq) => {
  const [open, setOpen] = useState(false)

  return (
    <>
      <div className='d-flex flex-column m-1'>
        <div className='d-flex flex-row justify-content-between'>
          <h3 className='fs-6'>{faq?.faq?.question}</h3>
          <button
            className='bg-transparent border-0'
            onClick={() => setOpen(!open)}
            aria-controls='example-collapse-text'
            aria-expanded={open}
          >
            {open ? (
              <i className='fa-solid fa-angle-down' />
            ) : (
              <i className='fa-solid fa-angle-up' />
            )}
          </button>
        </div>
        <Collapse in={open}>
          <div id='example-collapse-text' className='paragraph'>
            {faq?.faq?.answer}
          </div>
        </Collapse>
        {faq?.faq?.line && (
          <div className='col-12'>
            <hr className='mt-10' />
          </div>
        )}
      </div>
    </>
  )
}

export default FAQCard
