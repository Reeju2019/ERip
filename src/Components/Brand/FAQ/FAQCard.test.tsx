import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import FAQCard from './FAQCard'

interface FAQ {
  question: string
  answer: string
  line: boolean
}
const faq:FAQ = {
  question: "",
  answer: "",
  line: true
}
test('FAQ card', () => {
  render(<FAQCard faq={faq}/>)
})

describe('App rendering specification', () => {
  it('SnapShot testing', () => {
    const component = renderer.create(<FAQCard faq={faq}/>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('render without crashing', () => {
    const div = document.createElement('div')
    render(<FAQCard faq={faq}/>)
    ReactDOM.unmountComponentAtNode(div)
  })
})
