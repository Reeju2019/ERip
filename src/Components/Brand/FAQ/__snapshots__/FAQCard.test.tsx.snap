// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendering specification SnapShot testing 1`] = `
<div
  className="d-flex flex-column m-1"
>
  <div
    className="d-flex flex-row justify-content-between"
  >
    <h3
      className="fs-6"
    />
    <button
      aria-controls="example-collapse-text"
      aria-expanded={false}
      className="bg-transparent border-0"
      onClick={[Function]}
    >
      <i
        className="fa-solid fa-angle-up"
      />
    </button>
  </div>
  <div
    aria-expanded={null}
    className="paragraph collapse"
    id="example-collapse-text"
  />
  <div
    className="col-12"
  >
    <hr
      className="mt-10"
    />
  </div>
</div>
`;
